src/execute_linter.cc

```diff
@@ -24,24 +25,172 @@
 #include "absl/strings/string_view.h"
 #include "src/lint_errors.h"
 #include "src/linter.h"
+#include "src/linter_options.h"
 #include "zetasql/base/status.h"
 #include "zetasql/base/status_macros.h"
 
 namespace zetasql::linter {
 
 namespace {
+
+// This function will parse "NOLINT"(<CheckName>) syntax and adjust enability
+// options for each check. If NOLINT usage errors occur, it counts as a lint
+// error and it will be returned in a result.
+// This works similar to consistent comment type check.
+LinterResult ParseNoLintComments(absl::string_view sql, LinterOptions options,
+                                 std::map<ErrorCode, LinterOptions>* nolint) {
```

Comment:
Actually, please add a documentation what is the output inside 'nolint'. I don't really understand the idea...

Reply:
Bad naming. It is just a mapping between errorcode and options.